src/Workspaces/Remote/Core/Telemetry/ApiUsageIncrementalAnalyzerProvider.cs

```diff
@@ -121,9 +123,14 @@ public async Task AnalyzeProjectAsync(Project project, bool semanticsChanged, In
                 {
                     if (_reported.Add(project.Id))
                     {
+                        var solutionSessionId = project.Solution.State.SolutionAttributes.TelemetryId.ToString("B");
+                        var projectGuid = project.State.ProjectInfo.Attributes.TelemetryId.ToString("B");
+
                         // use telemetry API directly rather than Logger abstraction for PII data
                         var telemetryEvent = new TelemetryEvent(EventName);
-                        telemetryEvent.Properties[PropertyName] = new TelemetryComplexProperty(apiPerAssembly);
+                        telemetryEvent.Properties[ApiPropertyName] = new TelemetryComplexProperty(apiPerAssembly);
+                        telemetryEvent.Properties[SessionIdPropertyName] = new TelemetryPiiProperty(solutionSessionId);
+                        telemetryEvent.Properties[ProjectIdPropertyName] = new TelemetryPiiProperty(projectGuid);
```

Comment:
Is there some way we can ensure we don't accidentally pass these to a non-TelemetryPiiProperty? Should the SolutionAttributes and ProjectAttributes just not expose the underlying GUID and only expose a TelemetryPiiProperty?

Reply:
TelemetryPiiProperty is defined in a VS library which isn't available where the *Attributes are defined. I could store the hash, but then I lose the ability to see unhashed value while debugging locally.

Reply:
I think this is fine. Reporting the guids thru telemetry is the operation that should do the hashing/wrapping to TelemetryPiiProperty. There are many other pieces of data that are PII in the Roslyn solution (like the content of source files). 

---
In reply to: [376130036](https://github.com/dotnet/roslyn/pull/41443#discussion_r376130036) [](ancestors = 376130036)